s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py

```diff
@@ -0,0 +1,160 @@
+            'GET',
+            request_uri,
+            query_params,
+            body
+        )
+
+        if (headers['Authorization'] is None):
+            self._logger.error(fmt_reqid_log(self._request_id) +
+                               "Failed to generate v4 signature")
+            sys.exit(-1)
+
+        self._logger.info(fmt_reqid_log(self._request_id) +
+                          'GET on {}'.format(
+                              self._session.endpoint + request_uri))
+        self._logger.debug(fmt_reqid_log(self._request_id) +
+                           "GET Request Header {}".format(headers))
+
+        self._timer.start()
+        try:
+            async with self._session.get_client_session().get(
+                    self._session.endpoint + request_uri,
+                    params=query_params, headers=headers) as resp:
+
+                self._logger.info(
+                    fmt_reqid_log(self._request_id) +
+                    'GET response received with'
+                    + ' status code: {}'.format(resp.status))
+                self._logger.info('Response url {}'.format(
+                    self._session.endpoint + request_uri))
+
+                if resp.status == 200:
+                    self._response_headers = resp.headers
+                    received_tagset = await resp.text()
+                    self._logger.info(
+                        "Received tagset {}".format(received_tagset))
+
+                    # Remove namespace using regular expression
+                    # search and replace given pattern from the given string
+                    received_tagset = re.sub(
+                        'xmlns="[^"]+"', '', received_tagset)
+
+                    # Parse XML response
+                    root = fromstring(received_tagset)
+
+                    tags_dict = {}
+                    # Find all Tags elements in the entire tree.
+                    for ele in root.findall(".//Tag"):
+                        key = ele.find('Key').text
+                        value = ele.find('Value').text
+                        tags_dict[key] = value
+                    self._response_tags_dict = tags_dict
+
+                else:
+                    self._state = S3RequestState.FAILED
+                    error_msg = await resp.text()
+                    self._logger.error(
+                        fmt_reqid_log(self._request_id) +
+                        'GET failed with http status: {}'.
+                        format(resp.status) +
+                        'Error Response: {}'.format(error_msg))
+                    return
+
+        except aiohttp.client_exceptions.ClientConnectorError as e:
+            self.remote_down = True
+            self._state = S3RequestState.FAILED
+            self._logger.error(fmt_reqid_log(self._request_id) +
+                               "Failed to connect to S3: " + str(e))
+        self._timer.stop()
+        return
+
+    def pause(self):
```

Comment:
we will remove these calls in our api class implementations as it won't serve any purpose.

Reply:
done